bundles/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVBridgeHandler.java

```diff
@@ -0,0 +1,220 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.kvv.internal;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * KVVBridgeHandler encapsulates the communication with the KVV API.
+ *
+ * @author Maximilian Hess - Initial contribution
+ */
+@NonNullByDefault
+public class KVVBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(KVVBridgeHandler.class);
+
+    private final Cache cache;
+
+    private KVVBridgeConfig config;
+
+    private boolean wasOffline;
+
+    public KVVBridgeHandler(final Bridge bridge) {
+        super(bridge);
+        this.config = new KVVBridgeConfig();
+        this.cache = new Cache();
+        this.wasOffline = false;
+    }
+
+    public KVVBridgeConfig getBridgeConfig() {
+        return this.config;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        this.config = getConfigAs(KVVBridgeConfig.class);
+        if (this.config.apiKey == "") {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Failed to get bridge configuration");
+            return;
+        }
+
+        // test API by doing a dummy request to ensure it is available
+        final KVVStopConfig stopConfig = new KVVStopConfig();
+        stopConfig.stopId = "de:8212:6";
+        final DepartureResult departures = this.queryKVV(stopConfig);
+        if (departures == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Failed to connect to KVV API");
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // There is nothing to handle in the bridge handler
+    }
+
+    /**
+     * Returns the latest {@link DepartureResult}. Returns {@code null} if the result could not be retrieved.
+     *
+     * @return the latest {@link DepartureResult}.
+     */
+    public synchronized @Nullable DepartureResult queryKVV(final KVVStopConfig stopConfig) {
```

Comment:
This is actual a more relevant call for the stophandler. To beter model this you should put this method in a separate class, like `KVVQueryApi`. The bridge handler will get it passed and set the configuration. The stophandlers will get it passed and call it to get the specific stop data. The stop handler can just cache it in an `ExpiringCache` object and that should separate the logic better. The method `queryKVV` should throw an exception in case of errors, and the caller can catch on the exception and set the handler offline in case of error.